op-program/client/l2/engineapi/l2_engine_api.go

```diff
@@ -177,6 +177,31 @@ func (ea *L2EngineAPI) endBlock() (*types.Block, error) {
 }
 
 func (ea *L2EngineAPI) GetPayloadV1(ctx context.Context, payloadId eth.PayloadID) (*eth.ExecutionPayload, error) {
+	return ea.getPayload(ctx, payloadId)
+}
+
+func (ea *L2EngineAPI) GetPayloadV2(ctx context.Context, payloadId eth.PayloadID) (*eth.ExecutionPayloadEnvelope, error) {
+	payload, err := ea.getPayload(ctx, payloadId)
+	if err != nil {
+		return nil, err
+	} else {
+		return &eth.ExecutionPayloadEnvelope{ExecutionPayload: payload}, nil
+	}
+}
+
+func (ea *L2EngineAPI) ForkchoiceUpdatedV1(ctx context.Context, state *eth.ForkchoiceState, attr *eth.PayloadAttributes) (*eth.ForkchoiceUpdatedResult, error) {
+	return ea.forkchoiceUpdated(ctx, state, attr)
+}
+
```

Comment:
Is there any difference between fork choice updated v1 and v2 at all?

Reply:
I believe it's mostly validations on whether withdrawals are present or not (which I've just ported over).